src/entities/removeLiquidity/removeLiquidityCowAmm/index.ts

```diff
@@ -0,0 +1,141 @@
+import { Token } from '@/entities/token';
+import { TokenAmount } from '@/entities/tokenAmount';
+import { PoolState, PoolStateWithBalances } from '@/entities/types';
+import {
+    calculateProportionalAmounts,
+    getSortedTokens,
+} from '@/entities/utils';
+import { CHAINS } from '@/utils';
+
+import { getAmountsCall, getAmountsQuery } from '../helper';
+import {
+    RemoveLiquidityBase,
+    RemoveLiquidityBaseBuildCallInput,
+    RemoveLiquidityBaseQueryOutput,
+    RemoveLiquidityBuildCallOutput,
+    RemoveLiquidityKind,
+    RemoveLiquidityProportionalInput,
+} from '../types';
+import {
+    createPublicClient,
+    encodeFunctionData,
+    formatEther,
+    formatUnits,
+    http,
+} from 'viem';
+import { HumanAmount } from '@/data';
+import {
+    getPoolTokenBalanceCowAmm,
+    getTotalSupplyCowAmm,
+} from '@/entities/utils/cowAmmHelpers';
+import { cowAmmPoolAbi } from '@/abi/cowAmmPool';
+
+export class RemoveLiquidityCowAmm implements RemoveLiquidityBase {
+    public async query(
+        input: RemoveLiquidityProportionalInput,
+        poolState: PoolState,
+    ): Promise<RemoveLiquidityBaseQueryOutput> {
+        const sortedTokens = getSortedTokens(poolState.tokens, input.chainId);
+        const amounts = getAmountsQuery(sortedTokens, input);
+
+        const client = createPublicClient({
+            transport: http(input.rpcUrl),
+            chain: CHAINS[input.chainId],
+        });
+
+        const balances: bigint[] = [];
+        for (const token of sortedTokens) {
+            balances.push(
+                await getPoolTokenBalanceCowAmm(
```

Comment:
Similar to add about using multicall. Could probably have a common CowAMM helper like `getPoolState` that can be used for add and remove.

Reply:
done ✅